src/main/java/com/blamejared/crafttweaker/impl/managers/GenericRecipesManager.java

```diff
@@ -0,0 +1,177 @@
+        CraftTweakerAPI.apply(new ActionAddRecipe(recipeManagerWrapper, result, null));
+    }
+    
+    /**
+     * Removes recipes by output
+     *
+     * @param output The recipe result
+     * @docParam output <item:minecraft:iron_ingot>
+     */
+    @ZenCodeType.Method
+    public void removeRecipes(IItemStack output) {
+        CraftTweakerAPI.apply(new ActionRemoveFromWholeRegistryByOutput(output));
+    }
+    
+    /**
+     * Removes all recipes with this name.
+     * It is possible that there exist more than one recipe with the same name, if they exist in different recipe types.
+     *
+     * @param name The recipe name to remove
+     */
+    @ZenCodeType.Method
+    public void removeByName(String name) {
+        CraftTweakerAPI.apply(new ActionRemoveFromWholeRegistryByName(name));
+    }
+    
+    /**
+     * Removes all recipes from the provided mod.
+     * Chooses the recipes based on their full recipe name, not based on output item!
+     *
+     * @param modId The mod's modId
+     * @docParam modId "crafttweaker"
+     */
+    @ZenCodeType.Method
+    public void removeByModId(String modId) {
+        removeByModId(modId, null);
+    }
+    
+    /**
+     * Removes all recipes from the provided mod.
+     * Allows a function to exclude certain recipe names from being removed.
+     * In the example below, only the recipe for the white bed would remain.
+     * Since the recipe's namespace is already fixed based on the modId argument,
+     * the recipe filter will only check the resource path!
+     *
+     * @param modId   The mod's modid
+     * @param exclude Function that returns `true` if the recipe should remain in the registry.
+     * @docParam modId "minecraft"
+     * @docParam exclude (recipeName as string) => recipeName == "white_bed"
+     */
+    @ZenCodeType.Method
+    public void removeByModId(String modId, IRecipeManager.RecipeFilter exclude) {
+        CraftTweakerAPI.apply(new ActionRemoveFromWholeRegistryByModId(modId, exclude));
+    }
+    
+    /**
+     * Removes all recipes from all managers.
+     */
+    @ZenCodeType.Method
+    public void removeAll() {
+        CraftTweakerAPI.apply(new ActionRemoveFromWholeRegistryAllRecipes());
+    }
+    
+    /**
+     * Returns a list of all known recipe managers.
+     * This includes managers added by mod integrations as well as wrapper managers added to provide simple support.
+     */
+    @ZenCodeType.Method
+    @ZenCodeType.Getter("allManagers")
+    public List<IRecipeManager> getAllManagers() {
+        return new ArrayList<>(RecipeTypeBracketHandler.getManagerInstances());
+    }
+    
+    /**
+     * Returns a list of all known recipes.
+     * The map's value is a list, since it may be possible that two types have a recipe with the same name.
+     */
+    @ZenCodeType.Method
+    @ZenCodeType.Getter("allRecipes")
+    public Map<ResourceLocation, List<WrapperRecipe>> getAllRecipes() {
```

Comment:
@jaredlll08 
What do you think?
Technically, it's possible since they're all separate lists, but I doubt that there would be any case where this actually happens.
So, should I make it a List or a single Wrapper?

Reply:
What if we offered like a
```
recipes.recipeMap -> Map<ResourceLocation, List<WrapperRecipe>> 
```
and then

```
recipes.recipeList -> List<WrapperRecipe>
```
With the list just being a list of all the recipes?